apps/api/src/server/applications/application/documents/document.routes.js

```diff
@@ -17,6 +19,58 @@ import {
 
 const router = createRouter();
 
+router.get(
+	'/:id/documents/:guid/metadata',
+	/*
+        #swagger.tags = ['Applications']
+        #swagger.path = '/applications/{id}/documents/{guid}/metadata'
+        #swagger.description = 'This endpoint allows you to get metadata for a document associated with a case.'
+        #swagger.parameters['id'] = {
+            in: 'path',
+			description: 'Application ID here',
+			required: true,
+			type: 'integer'
+		}
+		#swagger.parameters['guid'] = {
+            in: 'path',
+			description: 'guid of the required document here',
+			required: true,
+			type: 'string'
+		}
+		#swagger.responses[200] = {
+			description: 'The metadata has returned successfully',
+			schema: { $ref: '#/definitions/storeMetadataResponse' }
+		}
+    */
+	asyncHandler(documentMetadata)
+);
+
+router.post(
+	'/:id/documents/:guid/metadata',
+	/*
+        #swagger.tags = ['Applications']
+        #swagger.path = '/applications/{id}/documents/{guid}/metadata'
+        #swagger.description = 'This endpoint allows you to store metadata for a document associated with a case.'
+        #swagger.parameters['id'] = {
+            in: 'path',
+			description: 'Application ID here',
+			required: true,
+			type: 'integer'
+		}
+		#swagger.parameters['guid'] = {
+            in: 'path',
+			description: 'guid of the required document here',
+			required: true,
+			type: 'string'
+		}
+		#swagger.responses[201] = {
+			description: 'The metadata has been successfully stored.',
+			schema: { $ref: '#/definitions/storeMetadataResponse' }
+		}
+    */
```

Comment:
Let's add a validation here for the data being sent over to the API and ensure only the expected parts arrive to the controller